server/modules/selva/module/selva_object.c

```diff
@@ -402,28 +430,16 @@ static int get_key_obj(struct SelvaObject *obj, const char *key_name_str, size_t
             (flags & SELVA_OBJECT_GETKEY_CREATE)) {
             /*
              * Either the nested object doesn't exist yet or the nested key is not an object,
-             * but we are allowed to create one here.
+             * but we are allowed to create one here. Only create the key if it didn't exist.
+             * Otherwise we can just reuse the old one.
              */
             if (!key) {
-                /*
-                 * Only create the key if it didn't exist. Otherwise we can just
-                 * reuse it.
-                 */
-                if (obj->obj_size == SELVA_OBJECT_SIZE_MAX) {
-                    return SELVA_OBJECT_EOBIG;
-                }
+                int err2;
```

Comment:
probably not necessary to have a new variable here but I didn't want to risk at this point 😓